src/components/Header/Header.tsx

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { useContext, useMemo, useState } from 'react';
+import cn from 'classnames';
+
+import { Errors } from '../../types/Errors';
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../../utils/TodosContext';
+import { ErrorContext } from '../../utils/ErrorContextProvider';
+
+interface Props {
+  onAdd: (val: Todo) => Promise<void>;
+  onToggleAll: () => void;
+  userId: number;
+}
+
+export const Header: React.FC<Props> = ({
+  userId,
+  onAdd,
+  onToggleAll,
+}) => {
+  const [title, setTitle] = useState('');
+
+  const { todos } = useContext(TodosContext);
+  const { showError } = useContext(ErrorContext);
+
+  const hasActiveTodos = useMemo(() => {
+    return todos.some(todo => !todo.completed);
+  }, [todos]);
+
+  const submitHandler = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!title.length) {
```

Comment:
trim it - I can add Todo with spaces only:
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/86923089/eaaa1c33-b741-4759-bddf-e38266681606)